nixos/modules/system/boot/networkd.nix

```diff
@@ -93,6 +93,28 @@ let
     (assertRange "FlowLabel" 0 1048575)
   ];
 
+  # NOTE The PrivateKey directive is missing on purpose here, please
+  # do not add it to this list. The nix store is world-readable let's
+  # refrain ourselves from providing a footgun.
+  checkWireguard = checkUnitConfig "WireGuard" [
+    (assertOnlyFields [
+      "PrivateKeyFile" "ListenPort" "FwMark"
+    ])
+    (assertInt "ListenPort")
+    (assertInt "FwMark")
```

Comment:
`ListenPort` takes either value between 1 and 65535 or `auto` [1]. `FwMark` takes a number between 1 and 4294967295 according to `man/systemd.netdev.xml` [2]. Also looks like it got renamed to `FirewallMark` but `FwMark` looks like it still exists [3].

[1] https://github.com/NixOS/systemd/blob/5c20aab77900f478fd380ab189787d80e4a35963/man/systemd.netdev.xml#L1254
[2] https://github.com/systemd/systemd/blob/master/man/systemd.netdev.xml#L1490
[3] https://github.com/systemd/systemd/blob/master/test/fuzz/fuzz-netdev-parser/directives.netdev#L13

Reply:
Just corrected the PR according to your remarks about `ListenPort` and the extra spaces.

As for `FwMark`, it seems that the current [pinned systemd](https://github.com/NixOS/systemd/blob/nixos-v242/man/systemd.netdev.xml#L1261) is not supporting the new syntax yet.